src/main/java/com/launchdarkly/eventsource/EventSource.java

```diff
@@ -59,17 +56,10 @@ public void run() {
   }
 
   public void stop() {
-      executor.shutdown();
-      try {
-        if (!executor.awaitTermination(5, TimeUnit.SECONDS)) {
-          executor.shutdownNow();
-          if (!executor.awaitTermination(5, TimeUnit.SECONDS))
-            System.err.println("Pool did not terminate");
-        }
-      } catch (InterruptedException ie) {
-        executor.shutdownNow();
-        Thread.currentThread().interrupt();
+      if (call != null) {
```

Comment:
I think I didn't do enough around lifecycle management to make this thread-safe. While I am usually a fan of finer-grained locking, perhaps this is a case where `synchronized` is enough?

I'll let you think through the right pattern to apply-- happy to chat it over as well.